README.md

```diff
@@ -700,4 +728,94 @@ calculateProportionalAmounts(
 ```
 
 Amounts proportional to the reference amount.
+___
+
+### Permit2 Helper
+
+Balancer v3 handles token approval through Pemit2 and this helper facilitates Permit2 signature generation.
+
+Each operation (e.g. addLiquidity, swap, ...) has its own method that leverages the same input type of the operation itself in order to simplify signature generation.
+
+**Example**
+
+See [addLiquidityWithPermit2Signature example](/examples/addLiquidity/addLiquidityWithPermit2Signature.ts).
+
+**Function**
+
+Helper function to create a Permit2 signature for an addLiquidity operation:
+
+```typescript
+static async signAddLiquidityApproval(
+    input: AddLiquidityBaseBuildCallInput & {
+        client: PublicWalletClient;
+        owner: Address;
+        nonces?: number[];
+        expirations?: number[];
+    },
+): Promise<Permit2>
+```
+
+**Parameters**
+
+| Name               | Type          | Description   |
+| -------------      | ------------- | ------------  |
+| input | [AddLiquidityBaseBuildCallInput](./src/entities/addLiquidity/types.ts#L62) | Add Liquidity Input |
+| client | [PublicWalletClient](./src/utils/types.ts#L3) | Viem's wallet client with public actions |
+| owner | Address | User address |
+| nonces (optional) | number[] | Nonces for each token |
+| expirations (optional) | number[] | Expirations for each token |
+
+**Returns**
+
+```typescript
+Promise<Permit2>;
+```
+
+[Permit2](./src/entities/permit2Helper/index.ts#L35) - Permit2 object with metadata and encoded signature
```

Comment:
I like this! Good balance between brevity and detail

If the plan is to detail each of the `Permit2Helper` methods, this is a great blueprint :muscle:

Reply:
i'm on the fence as well. might not be a bad idea to hope for the best and only add more explicit docs for each method if we get complaints